hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieWriterUtils.scala

```diff
@@ -115,7 +116,12 @@ object HoodieWriterUtils {
   }
 
   def getOriginKeyGenerator(parameters: Map[String, String]): String = {
-    val kg = parameters.getOrElse(KEYGENERATOR_CLASS_NAME.key(), null)
+    //first check table config for key generator
+    var kg = parameters.getOrElse(HoodieTableConfig.KEY_GENERATOR_CLASS_NAME.key, null)
```

Comment:
kg -> keyGenClass

Reply:
why can't we directly do getOrDefault in L120?

Reply:
You mean like this ?

var kg = parameters.getOrElse(HoodieTableConfig.KEY_GENERATOR_CLASS_NAME.key, parameters.getOrElse(KEYGENERATOR_CLASS_NAME.key(), KEYGENERATOR_CLASS_NAME.defaultValue()))

The only drawback to this is it might affect readability.

Reply:
yes. but getOrElse or getOrDefault is used widely across the code base. should be ok

Reply:
This code has been reverted since we dont need to change the logic here.